tests/js/searchtools.js

```diff
@@ -1,16 +1,19 @@
 describe('Basic html theme search', function() {
 
+  function loadFixture(name) {
+      req = new XMLHttpRequest();
+      req.open("GET", `base/tests/js/fixtures/${name}`, false);
+      req.send(null);
+      return req.responseText;
+  }
+
   describe('terms search', function() {
 
     it('should find "C++" when in index', function() {
-      index = {
-        docnames:["index"],
-        filenames:["index.rst"],
-        terms:{'c++':0},
-        titles:["&lt;no title&gt;"],
-        titleterms:{}
-      }
-      Search.setIndex(index);
+      searchindex = loadFixture("index.cpp.js");
```

Comment:
Ah, hm.  I have a suggestion for myself here: load these using the filename `searchindex.js`, but from a directory that has the same name as the source directory under `js/roots`.

That'll be clearer and easier when mapping the source artifacts to their destinations.